plugin/documents.py

```diff
@@ -285,6 +287,12 @@ def on_diagnostics_updated_async(self, is_view_visible: bool) -> None:
         is_active_view = window and window.active_view() == self.view
         if is_active_view and self.view.change_count() == self._change_count_on_last_save:
             self._toggle_diagnostics_panel_if_needed_async()
+        self._diagnostics_view.clear_annotations()
+        if 'annotation' in userprefs().show_diagnostics:
+            all_diagnostics = []  # type: List[Tuple[Diagnostic, sublime.Region]]
+            for _, diagnostics in self._diagnostics_async(allow_stale=True):
+                all_diagnostics.extend(diagnostics)
+            self._diagnostics_view.update_diagnostic_annotations_async(all_diagnostics)
```

Comment:
Just realized that this should be tied to `SessionView` and and `DocumentSyncListener`. Otherwise it won't work in split views.

Reply:
No, never mind, the current code is correct. It's just there is issue with ST that the LSP doesn't immediately initialize on splitting the view. Something needs to change in the file for it to trigger.
Sounds like one of the issues we had but not sure if it's a known issue, it regressed or it's a new one.